akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -66,6 +68,18 @@ private[io] final class AsyncDnsResolver(
     }
   }
 
+  private def alreadyResolvedIp(knownToBeIp: String): Future[DnsProtocol.Resolved] = {
+    if (isInetAddress(knownToBeIp))
+      Future {
```

Comment:
Why execute this in a `Future` if we know that no async processing is done?

Reply:
Since I assumed the `val address = InetAddress.getByName(knownToBeIp) // only checks validity, since known to be IP address` can throw so I'd want to expose that as the future failure, thus could not use `Future.successful`. We could ride on the assumption it should not throw since we validated the IP however likely better not to.

Reply:
I see, we could then use `Future.fromTry`.

Reply:
Ah, great reminder thanks!